lib/metasploit/framework/login_scanner/opnsense.rb

```diff
@@ -0,0 +1,141 @@
+require 'metasploit/framework/login_scanner/http'
+
+module Metasploit
+  module Framework
+    module LoginScanner
+
+      # This is the LoginScanner class for dealing with Deciso B.V. OPNSense instances.
+      # It is responsible for taking a single target, and a list of credentials
+      # and attempting them. It then saves the results.
+      class OPNSense < HTTP
+
+        def get_cookie_value(response, wanted_cookie_name)
+          response.get_cookies.split('; ').find { |cookie| cookie.start_with?(wanted_cookie_name) }.split('=').last
+        end
+
+        # Sends a HTTP request with Rex
+        #
+        # @param (see Rex::Proto::Http::Request#request_raw)
+        # @return [Rex::Proto::Http::Response] The HTTP response
+        def send_request(opts, keep_cookies = false)
+          res = super(opts)
+
+          if keep_cookies && res
+            @php_sessid = get_cookie_value(res, 'PHPSESSID')
+            @cookie_test = get_cookie_value(res, 'cookie_test')
+          end
+
+          res
+        end
+
+        # include Msf::Exploit::Remote::HTTP::OPNSense::Login
+        # include Msf::Exploit::Remote::HTTP::HttpClient
+        # define_method :send_request_cgi, Msf::Exploit::Remote::HttpClient.instance_method(:send_request_cgi)
+        # Checks if the target is OPNSense. The login module should call this.
+        #
+        # @return [Boolean, String] FalseClass if target is OPNSense, otherwise String
+        def check_setup
+          request_params = {
+            'method' => 'GET',
+            'uri' => normalize_uri(@uri.to_s)
+          }
+          res = send_request(request_params)
+
+          if res && res.code == 200 && res.body&.include?('Login | OPNsense')
+            return false
+          end
+
+          "Unable to locate \"Login | OPNsense\" in body. (Is this really OPNSense?)"
+        end
+
+        def query_magic_value
```

Comment:
Would you be able to add a method definition here? A brief description of the hash being returned would be great.